warehouse/ingest-core/src/main/java/datawave/ingest/util/NGramTokenizationStrategy.java

```diff
@@ -131,48 +127,33 @@ public int tokenize(final NormalizedContentInterface content, int maxNGramLength
                     }
 
                     // Create the tokenizer and tokenizing stream
-                    tokenizer = new NGramTokenizer(2, maxNGramLength);
-                    tokenizer.setReader(reader);
-                    tokenizer.reset();
-                    tokenStream = new ClassicFilter(tokenizer);
-                    tokenStream = new LowerCaseFilter(tokenStream);
-                    tokenStream.addAttribute(CharTermAttribute.class);
-
-                    // Reset the n-gram count
-                    ngramCount = 0;
-
-                    // Increment the tokenizer and applied any generated n-grams
-                    String ngram = null;
-                    while (null != (ngram = this.increment(tokenizer))) {
-                        if (this.updateFilter(ngram, content)) {
-                            ngramCount++;
+                    try (NGramTokenizer tokenizer = new NGramTokenizer(2, maxNGramLength)) {
+                        tokenizer.setReader(reader);
+                        tokenizer.reset();
+                        try (TokenStream tokenStream = new ClassicFilter(tokenizer)) {
+                            try (TokenStream lcfTokenStream = new LowerCaseFilter(tokenStream)) {
+                                lcfTokenStream.addAttribute(CharTermAttribute.class);
+
+                                // Reset the n-gram count
+                                ngramCount = 0;
+
+                                // Increment the tokenizer and applied any generated n-grams
+                                String ngram = null;
+                                while (null != (ngram = this.increment(tokenizer))) {
+                                    if (this.updateFilter(ngram, content)) {
+                                        ngramCount++;
+                                    }
+                                }
+                            }
+                            if (null != tokenizer) {
```

Comment:
Since we called `tokenizer.setReader` and `tokenizer.reset` above, it looks like this null check is not needed.

Reply:
true - removed.